src/diet/index.ts

```diff
@@ -0,0 +1,506 @@
+      if (item.notes?.includes("MARTINI") && modifiers.tuxedoShirt) {
+        adventures += 2;
+      }
+      if (have($skill`Saucemaven`) && item.notes?.includes("SAUCY")) {
+        adventures += myPrimestat() === $stat`Mysticality` ? 5 : 3;
+      }
+      if (itemType(item) === "food" && modifiers.seasoning) adventures++;
+      if (itemType(item) === "food" && modifiers.mayoflex) adventures++;
+      return adventures;
+    }) / interpolated.length
+  );
+}
+
+type MenuItemOptions = {
+  organ?: Organ;
+  size?: number;
+  maximum?: number | "auto";
+  additionalValue?: number;
+  wishEffect?: Effect;
+};
+
+export class MenuItem {
+  item: Item;
+  organ?: Organ;
+  size: number;
+  maximum?: number;
+  additionalValue?: number;
+  wishEffect?: Effect;
+
+  static defaultOptions = new Map([
+    [$item`Mr. Burnsger`, { maximum: "auto" }],
+    [
+      $item`distention pill`,
+      {
+        organ: "food",
+        maximum: "auto",
+        size: -1,
+      },
+    ],
+    [
+      $item`synthetic dog hair pill`,
+      { organ: "booze", maximum: "auto", size: -1 },
+    ],
+    [$item`cuppa Voraci tea`, { organ: "food", maximum: "auto", size: -1 }],
+    [$item`cuppa Sobrie tea`, { organ: "booze", maximum: "auto", size: -1 }],
+    [$item`mojo filter`, { organ: "spleen item", maximum: "auto", size: -1 }],
+  ] as [Item, MenuItemOptions][]);
+
+  constructor(item: Item, options: MenuItemOptions = {}) {
+    const { size, organ, maximum, additionalValue, wishEffect } = {
+      ...options,
+      ...(MenuItem.defaultOptions.get(item) ?? {}),
+    };
+    this.item = item;
+    this.maximum = maximum === "auto" ? item.dailyusesleft : maximum;
+    this.additionalValue = additionalValue;
+    this.wishEffect = wishEffect;
+
+    const typ = itemType(this.item);
+    this.organ = organ ?? (isOrgan(typ) ? typ : undefined);
+    this.size =
+      size ??
+      (this.organ === "food"
+        ? this.item.fullness
+        : this.organ === "booze"
+        ? this.item.inebriety
+        : this.organ === "spleen item"
+        ? this.item.spleen
+        : 0);
+  }
+
+  equals(other: MenuItem) {
+    return this.item === other.item && this.wishEffect === other.wishEffect;
+  }
+
+  toString(): string {
+    return this.item.toString();
+  }
+
+  price(): number {
+    return npcPrice(this.item) > 0 ? npcPrice(this.item) : mallPrice(this.item);
+  }
+}
+
+const organs = ["food", "booze", "spleen item"] as const;
+type Organ = typeof organs[number];
+type OrganSize = [Organ, number];
+
+function isOrgan(x: string): x is Organ {
+  return (organs as readonly string[]).includes(x);
+}
+
+class DietPlanner {
+  mpa: number;
+  menu: MenuItem[];
+  fork?: MenuItem;
+  mug?: MenuItem;
+  seasoning?: MenuItem;
+  mayoflex?: MenuItem;
+  pinkyRing: boolean;
+  tuxedoShirt: boolean;
+  spleenValue = 0;
+
+  constructor(mpa: number, menu: MenuItem[]) {
+    this.mpa = mpa;
+    this.fork = menu.find(
```

Comment:
It may be wise to not explicitly define these as undefined, because that can make object spread syntax different for reasons that elude me. So you could do something like
```
const mug =  menu.find((item) => item.item === $item`Frosty's frosty mug`);
if (mug) this.mug = mug;
```

Reply:
I don't think this matters, these properties aren't enumerable anyway